processor/src/main/java/org/mapstruct/ap/spi/PresenceCheckAccessorNamingStrategy.java

```diff
@@ -0,0 +1,53 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.spi;
+
+
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.type.TypeKind;
+
+/**
+ * The default JavaBeans-compliant implementation of the {@link AccessorNamingStrategy} service provider interface.
+ *
+ * @author Sjaak Derksen
+ */
+public class PresenceCheckAccessorNamingStrategy
```

Comment:
I think that that came from one of your previous comments (on @seanjob 's implementation).. But I'm not sure anymore. I think you did not want to make a definite choice for `has`.

So: in the current solution: even if you implement `has` methods, they will not be used for presence checking unless you make a deliberate choice. In that case: you have one predefined.